src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -46,7 +49,7 @@ private static AbstractManagedIdentity SelectManagedIdentitySource(RequestContex
         // Detect managed identity source based on the availability of environment variables.
         // The result of this method is not cached because reading environment variables is cheap. 
         // This method is perf sensitive any changes should be benchmarked.
-        internal static ManagedIdentitySource GetManagedIdentitySource()
+        internal static ManagedIdentitySource GetManagedIdentitySource(ILoggerAdapter logger = null)
```

Comment:
not a fan of this "maybe null" approach for objects. Can't the product code guarantee it is non-null all the time? Tests can inject mocks if needed.

Reply:
This is not for the tests. This method is called from 2 places. One when the request context and everything is available and the other is a static method from ManagedIdentityApplication. The logger is not available at that point.